### PR TITLE
[5.1] Fews improvement to Routing component:

### PR DESCRIPTION
**Pull request by crynobone**
- Avoid using `last()` since it's just an alias to `end()`
- Refactor `array_get()` to just use `isset($a) ? $a : null` when possible.
- Avoid using `array_get()` on middleware name to allow using '.' as route middleware name.

Signed-off-by: crynobone crynobone@gmail.com

### CODE REVIEW

File: src/Illuminate/Routing/Router.php [hunk 1]
```diff
@@ -719,7 +724,7 @@ public function resolveMiddlewareClassName($name)
 
         list($name, $parameters) = array_pad(explode(':', $name, 2), 2, null);
 
-        return array_get($map, $name, $name).($parameters ? ':'.$parameters : '');
+        return (isset($map[$name]) ? $map[$name] : $name).($parameters ? ':'.$parameters : '');
```
Comment (crynobone) [inline src/Illuminate/Routing/Router.php]: Avoid using `array_get()` on middleware name to allow using '.' as route middleware name.

**Comment by franzliedke** [comment]
Good stuff!

File: src/Illuminate/Routing/Router.php [hunk 2]
```diff
@@ -406,12 +409,12 @@ public static function mergeGroup($new, $old)
     protected static function formatUsesPrefix($new, $old)
     {
         if (isset($new['namespace']) && isset($old['namespace'])) {
-            return trim(array_get($old, 'namespace'), '\\').'\\'.trim($new['namespace'], '\\');
+            return trim($old['namespace'], '\\').'\\'.trim($new['namespace'], '\\');
```
Comment (crynobone) [inline src/Illuminate/Routing/Router.php]: The condition already check for `isset()`, `array_get()` here is just redundant.

**Comment by crynobone** [comment]
Need to fix some bug, will resend later.

**Pull request closed without merging**